### PR TITLE
Add search bars

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it adds search ability to most of the list views.

### Screenshots
Screenshot 2019-09-12 at 14 44 46

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted to `.pot` file.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by maarcingebala** [comment]
We should use `{filter: search: ""}` arguments in all views. E.g. Orders page doesn't use it now.

File: src/components/Filter/FilterActions.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    />
+  );
+};
+
+const useStyles = makeStyles(
+  (theme: Theme) => ({
+    actionContainer: {
+      display: "flex",
+      flexWrap: "wrap",
+      padding: `${theme.spacing.unit * 1.5}px ${theme.spacing.unit * 3}px ${
```
Comment (benekex2) [inline src/components/Filter/FilterActions.tsx]: You use one time `px` value one time `theme.spacing.unit`. We should set the correct pattern.

File: src/components/Filter/FilterSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  onSearchSave?: () => void;
+}
+
+const useStyles = makeStyles(
+  (theme: Theme) => ({
+    tabAction: {
+      display: "inline-block"
+    },
+    tabActionContainer: {
+      borderBottom: "1px solid #e8e8e8",
```
Comment (benekex2) [inline src/components/Filter/FilterSearch.tsx]: Use variable

File: src/components/TableFilter/FilterChips.tsx [hunk 3]
```diff
@@ -17,98 +17,64 @@ export interface Filter {
...
+    filterButton: {
+      alignItems: "center",
+      backgroundColor: fade(theme.palette.primary.main, 0.6),
+      borderRadius: "19px",
+      display: "flex",
+      height: "38px",
+      justifyContent: "space-around",
+      margin: `0 ${theme.spacing.unit * 2}px ${theme.spacing.unit}px`,
+      marginLeft: 0,
+      padding: "0 16px"
```
Comment (benekex2) [inline src/components/TableFilter/FilterChips.tsx]: Same as above

File: src/components/TableFilter/FilterChips.tsx [hunk 4]
```diff
@@ -17,98 +17,64 @@ export interface Filter {
...
+    },
+    filterChipContainer: {
+      display: "flex",
+      flex: 1,
+      flexWrap: "wrap"
+    },
+    filterContainer: {
+      borderBottom: "1px solid #e8e8e8",
```
Comment (benekex2) [inline src/components/TableFilter/FilterChips.tsx]: Same as above

File: src/attributes/views/AttributeList/AttributeList.tsx [hunk 5]
```diff
@@ -37,6 +50,15 @@ const AttributeList: React.FC<AttributeListProps> = ({ params }) => {
   );
   const intl = useIntl();
 
+  const tabs = getFilterTabs();
+
+  const currentTab =
```
Comment (benekex2) [inline src/attributes/views/AttributeList/AttributeList.tsx]: You can extract this as separate util

File: src/attributes/views/AttributeList/AttributeList.tsx [hunk 6]
```diff
@@ -37,6 +50,15 @@ const AttributeList: React.FC<AttributeListProps> = ({ params }) => {
...
+    params.activeTab === undefined
+      ? areFiltersApplied(params)
+        ? tabs.length + 1
+        : 0
+      : parseInt(params.activeTab, 0);
+
   const closeModal = () =>
```
Comment (benekex2) [inline src/attributes/views/AttributeList/AttributeList.tsx]: You can extract this as separate util

File: src/attributes/views/AttributeList/AttributeList.tsx [hunk 7]
```diff
@@ -56,8 +78,46 @@ const AttributeList: React.FC<AttributeListProps> = ({ params }) => {
       })
     );
 
+  const changeFilterField = (filter: AttributeListUrlFilters) => {
```
Comment (benekex2) [inline src/attributes/views/AttributeList/AttributeList.tsx]: You can extract this as separate util

File: src/attributes/views/AttributeList/AttributeList.tsx [hunk 8]
```diff
@@ -56,8 +78,46 @@ const AttributeList: React.FC<AttributeListProps> = ({ params }) => {
...
+    navigate(
+      attributeListUrl({
+        ...getActiveFilters(params),
+        ...filter,
+        activeTab: undefined
+      })
+    );
+  };
+
+  const handleTabChange = (tab: number) => {
```
Comment (benekex2) [inline src/attributes/views/AttributeList/AttributeList.tsx]: You can extract this as separate util

File: src/attributes/views/AttributeList/AttributeList.tsx [hunk 9]
```diff
@@ -56,8 +78,46 @@ const AttributeList: React.FC<AttributeListProps> = ({ params }) => {
...
+    reset();
+    navigate(
+      attributeListUrl({
+        activeTab: tab.toString(),
+        ...getFilterTabs()[tab - 1].data
+      })
+    );
+  };
+
+  const handleTabDelete = () => {
```
Comment (benekex2) [inline src/attributes/views/AttributeList/AttributeList.tsx]: You can extract this as separate util

File: src/attributes/views/AttributeList/AttributeList.tsx [hunk 10]
```diff
@@ -56,8 +78,46 @@ const AttributeList: React.FC<AttributeListProps> = ({ params }) => {
...
+    deleteFilterTab(currentTab);
+    reset();
+    navigate(attributeListUrl());
+  };
+
+  const handleTabSave = (data: SaveFilterTabDialogFormData) => {
```
Comment (benekex2) [inline src/attributes/views/AttributeList/AttributeList.tsx]: You can extract this as separate util

**Pull request merged**